examples/connectivity/plot_extract_regions_canica_maps.py

```diff
@@ -0,0 +1,139 @@
+and :class:`nilearn.decomposition.CanICA` for whole brain ICA maps.
+
+Please see the related documentation of :class:`nilearn.regions.RegionExtractor`
+for more details.
+"""
+
+################################################################################
+# Fetching ADHD resting state functional datasets by loading from datasets
+# utilities
+from nilearn import datasets
+
+adhd_dataset = datasets.fetch_adhd(n_subjects=20)
+func_filenames = adhd_dataset.func
+confounds = adhd_dataset.confounds
+
+################################################################################
+# Canonical ICA decomposition of functional datasets by importing CanICA from
+# decomposition module
+from nilearn.decomposition import CanICA
+
+# Initialize canica parameters
+n_components = 5
+canica = CanICA(n_components=n_components, smoothing_fwhm=6.,
+                memory="nilearn_cache", memory_level=2,
+                random_state=0)
+# Fit to the data
+canica.fit(func_filenames)
+# ICA maps
+components_img = canica.masker_.inverse_transform(canica.components_)
+
+# Visualization
+# Show ICA maps by using plotting utilities
+import matplotlib.pyplot as plt
+from nilearn import plotting
+
+fig1 = plt.figure(figsize=(10, 5))
+plotting.plot_prob_atlas(components_img, view_type='filled_contours',
+                         figure=fig1, title='ICA components')
+
+################################################################################
+# Extracting regions from ICA maps and then timeseries signals from those
+# regions, both can be done by importing Region Extractor from regions module
+from nilearn.regions import RegionExtractor
+
+extractor = RegionExtractor(components_img, threshold=0.5,
+                            thresholding_strategy='ratio_n_voxels',
+                            standardize=True, min_region_size=50)
+# Just call fit() to process for regions extraction
+extractor.fit()
+# Extracted regions are stored in regions_img_
+regions_extracted_img = extractor.regions_img_
+
+# Visualization
+# Show region extraction results
+fig2 = plt.figure(figsize=(10, 5))
+title = ('%d regions are extracted from %d ICA components. '
+         '\nEach separate color of region indicates extracted region'
+         % (regions_extracted_img.shape[-1], n_components))
+plotting.plot_prob_atlas(regions_extracted_img, view_type='filled_contours',
+                         title=title, figure=fig2)
+
+################################################################################
+# Computing correlation coefficients
+# First step is to do subjects timeseries signals extraction
+subjects_timeseries = []
+for img, confound in zip(func_filenames, confounds):
+    signals = extractor.transform(img, confounds=confound)
+    subjects_timeseries.append(signals)
+
+# Import numpy to compute correlation and mean
+import numpy as np
+
+# Second step is to compute correlation coefficients
+correlations = [np.corrcoef(subject_ts.T) for subject_ts in subjects_timeseries]
```

Comment:
In the reference documentation, I gave a reference to ConnectivityMeasure in nilearn saying that we have ready to compute object for functional connectivity measures.